src/Controls/src/Core/Handlers/Items/ItemsViewHandler.Android.cs

```diff
@@ -20,9 +18,36 @@ protected override void ConnectHandler(RecyclerView platformView)
 			base.ConnectHandler(platformView);
 			(platformView as IMauiRecyclerView<TItemsView>)?.SetUpNewElement(VirtualView);
 		}
+
 		protected override RecyclerView CreatePlatformView() =>
 			new MauiRecyclerView<TItemsView, ItemsViewAdapter<TItemsView, IItemsViewSource>, IItemsViewSource>(Context, GetItemsLayout, CreateAdapter);
 
+		public override Size GetDesiredSize(double widthConstraint, double heightConstraint)
+		{
+			var width = widthConstraint;
+			var height = heightConstraint;
+
+			if (!double.IsInfinity(width))
+				width = Context.ToPixels(width);
+
+			if (!double.IsInfinity(height))
+				height = Context.ToPixels(height);
+
+			UpdateEmptyViewSize(width, height);
```

Comment:
Won't this try to force the RecyclerView to always grow to the maximum constraint size?  There's an issue on iOS where this already seems to happen and we want to make the behaviour more like Android to only grow to the size of its content if that does not exceed the constraints.

Reply:
```suggestion
            var result = base.GetDesiredSize(widthConstraint, heightConstraint);
			UpdateEmptyViewSize(result.Width, result.Height);
			return result;
```

Reply:
Not a perfect solution, but it'll prevent the issue @redth is worried about.